controllers/yawol-controller/loadbalancer/loadbalancer_controller_test.go

```diff
@@ -248,6 +250,93 @@ var _ = Describe("loadbalancer controller", Serial, Ordered, func() {
 			})
 		})
 
+		It("should scale down old lbset after new one has ready keepalived", func() {
+			var oldLbs runtimeClient.ObjectKey
+			By("waiting for lbset creation")
+			hopefully(lbNN, func(g Gomega, act LB) error {
+				var lbsetList yawolv1beta1.LoadBalancerSetList
+				g.Expect(k8sClient.List(ctx, &lbsetList, &runtimeClient.ListOptions{
+					LabelSelector: labels.SelectorFromSet(lb.Spec.Selector.MatchLabels),
+				})).Should(Succeed())
+
+				g.Expect(len(lbsetList.Items)).Should(Equal(1))
+				oldLbs = runtimeClient.ObjectKeyFromObject(&lbsetList.Items[0])
+				return nil
+			})
+
+			By("changing flavorid")
+			updateLB(lbNN, func(act *LB) {
+				act.Spec.Infrastructure.Flavor = yawolv1beta1.OpenstackFlavorRef{
+					FlavorID: pointer.String("somenewid"),
+				}
+			})
+
+			var newLbs runtimeClient.ObjectKey
+			By("checking for a new lbset")
+			hopefully(lbNN, func(g Gomega, act LB) error {
+				var lbsetList yawolv1beta1.LoadBalancerSetList
+				g.Expect(k8sClient.List(ctx, &lbsetList, &runtimeClient.ListOptions{
+					LabelSelector: labels.SelectorFromSet(lb.Spec.Selector.MatchLabels),
+				})).Should(Succeed())
+
+				g.Expect(len(lbsetList.Items)).Should(Equal(2))
+				for i := range lbsetList.Items {
+					if lbsetList.Items[i].Annotations[helper.RevisionAnnotation] == "2" {
+						newLbs = runtimeClient.ObjectKeyFromObject(&lbsetList.Items[i])
+					}
+				}
+				return nil
+			})
+
+			By("Make both lbsets available by path status", func() {
+				var lbsetList yawolv1beta1.LoadBalancerSetList
+				Expect(k8sClient.List(ctx, &lbsetList, &runtimeClient.ListOptions{
+					LabelSelector: labels.SelectorFromSet(lb.Spec.Selector.MatchLabels),
+				})).Should(Succeed())
+				for _, lbs := range lbsetList.Items {
+					lbs.Status.ReadyReplicas = &lbs.Spec.Replicas
+					lbs.Status.Replicas = &lbs.Spec.Replicas
+					Expect(k8sClient.Status().Update(ctx, &lbs)).Should(Succeed())
+				}
+			})
+
+			By("Old lbset should be scaled up because keepalived is not ready now")
+			Consistently(func() error {
+				var lbs yawolv1beta1.LoadBalancerSet
+				if err := k8sClient.Get(ctx, oldLbs, &lbs); err != nil {
+					return err
+				}
+				if lbs.Spec.Replicas == 0 {
+					return errors.New("already down scaled")
+				}
+
+				return nil
+			}, time.Second*8).Should(Succeed())
+
+			By("Make latest lbsets keepalived condition true by path status", func() {
+				var lbs yawolv1beta1.LoadBalancerSet
+				Expect(k8sClient.Get(ctx, newLbs, &lbs)).Should(Succeed())
+				lbs.Status.Conditions = []metav1.Condition{
+					{
+						Type:               helper.ContainsKeepalivedMaster,
+						Status:             metav1.ConditionTrue,
+						LastTransitionTime: metav1.Time{Time: metav1.Now().Add(-120 * time.Second)},
```

Comment:
With this, we immediately cross the threshold and expect the old set to be scaled down.
Can we do this in two steps (closer to reality)? I.e., first make the set contain the master (since 0s), expect both sets to have a replica, and only then update the transition time and expect that the old one is scaled down.